fix(source-intercom): incremental not emitting state

darynaishchenko opened the pull request:
## What
https://github.com/airbytehq/oncall/issues/7302
cdk doesn’t emit state while reading records, because checkpoint_interval for declarative stream  is None. And in the end of the read cdk also doesn’t emit state because new_state == previous_state.

## How
fixed `get_stream_state` to return a copy of current state not ref to the state. updated to latest cdk with fix for checkpointing

## Review guide
<!--
1. `x.py`
2. `y.py`
-->

## User Impact
<!--
* What is the end result perceived by the user?
* If there are negative side effects, please list them. 
-->

## Can this PR be safely reverted and rolled back?
<!--
* If unsure, leave it blank.
-->
- [x] YES 💚
- [ ] NO ❌

maxi297 commented:
@lazebnyi can you take this one?

darynaishchenko commented on airbyte-integrations/connectors/source-intercom/integration_tests/abnormal_state.json:
```diff
@@ -70,16 +70,5 @@
         "updated_at": 7626086649
       }
     }
-  },
```
### CAT failed due to error:
```
>                       assert isinstance(
                            actual_cursor_value, _type
                        ), f"Cursor value {actual_cursor_value} is not of type {_type}. Expected {_type}, got {type(actual_cursor_value)}"
E                       AssertionError: Cursor value 7626086649 is not of type <class 'int'>. Expected <class 'int'>, got <class 'str'>
E                       assert False
E                        +  where False = isinstance('7626086649', <class 'int'>)
```
### Reason
Activity logs stream is concurrent now. When [convert_to_state_message()](https://github.com/airbytehq/airbyte-python-cdk/blob/main/airbyte_cdk/sources/streams/concurrent/state_converters/abstract_stream_state_converter.py#L42) was called it converts provided state(int format) to string("%s") [here](https://github.com/airbytehq/airbyte-python-cdk/blob/main/airbyte_cdk/sources/declarative/datetime/datetime_parser.py#L28). Cursor field is integer in stream schema, that's why cat test was expected int value.

I removed this stream from testing for now. I don't expect this change will break current sync, as 1. cursor field didn't change its type in schema and records. 2. airbyte cdk converts state values to one format, works with both int and str.

@maxi297 what do you think?

maxi297 replied:
It should be related to https://airbytehq-team.slack.com/archives/C06P578H8J1/p1737473608736379?thread_ts=1737375056.932079&cid=C06P578H8J1. Can you confirm? If so, we can just silence the future_state tests

darynaishchenko commented:
**Regression test results:**
_run regression test locally, because using it via gha lead to error for connection version as we pinned v0.8.3_

**State count 0.9.0 vs dev:**
39	64	diff=25
dev version has more states as we fixed custom cursor and updated cdk to latest version with a fix for checkpointing.

### test_catalog_are_the_same[failed] 
full refresh streams are now not resumable, related on cdk update. Acceptable and expected.

### test_all_records_are_the_same_with_state[failed for when control 0.8.3, but passed when control is 0.9.0]
Expected: on latest cdk version records don't have some fields when their value Null

darynaishchenko commented:
/approve-regression-tests
